### DISCUSSION TITLE
Calibration of circuit sensors?

### DISCUSSION THREAD

**Discussion by TommySharpNZ** [emporia-vue-local/esphome#71] [Q&A]
I've tinkered and found that using "calibration: 0.02283201" for "phase_a" gives me a reading that is pretty much equivalent to what the power company meter is showing on it's LCD.

I also have an OpenEVSE car charger and it's internal CT clamp and power measurement says it's charging my car at 3,590w but the circuit power reported by esphome on my Emporia is 3,265w so quite a difference...

What is the best approach to work out which is actually correct and how to calibrate if necessary?

**Comment by flaviut** [comment]
The way this is done in industry is by having test equipment that has been calibrated against standards at a national lab like NIST. No idea what pricing for this is like, you'll need to request a quote from a calibration lab. ([for example](https://www.tek.com/en/services/calibration-services))

As far as what to do for your home, I would trust your electric meter. There are legal requirements for some level of accuracy, while there is no such requirement for your EV charger.

**Reply by TommySharpNZ** [reply]
I'm just wondering if maybe I need to apply a calibration value to the circuit CT clamp that is connected to the Emporia?
My car has a 3.6kw charger in it so I kinda wonder if the 3.265kw being reported by the Emporia circuit CT clamp is a bit low?

**Reply by flaviut** [reply]
> I'm just wondering if maybe I need to apply a calibration value to the circuit CT clamp that is connected to the Emporia?

I see! Yes, you can. That's not a bad idea.

>  My car has a 3.6kw charger in it so I kinda wonder if the 3.265kw being reported by the Emporia circuit CT clamp is a bit low?

No way to tell who is right without putting a good quality multi-meter in the circuit unfortunately. Fluke or similar, this is scary enough I would stay away from anything cheap.

**Reply by clowrey** [reply]
Any AC clamp multimeter will give you another data point to calibrate against. Even the low cost ones are fairly accurate.

**Reply by fhloston** [reply]
As a comparable device like the Shelly 3EM comes factory calibrated and this calibration data can be exported before flashing another firmware to it - is it possible the Emporia Vue is also calibrated and has this as exportable data?

**Reply by flaviut** [reply]
> is it possible the Emporia Vue is also calibrated

No. I've dug deeply into this while reverse engineering it.

In addition, the calibration would be most accurate when a particular clamp is only used with a particular channel. There's no such markers on the Vue.

There are calibration constants for voltage in the firmware that I have been unable to figure out, but those are related to the order that the voltage wires are connected.

**Reply by fhloston** [reply]
Thank you, makes sense. None of the Emporia sensors are marked in contrast to the sensors on the Shelly 3EM which are clearly labeled and calibrated.
